jolt-core/src/jolt/vm/mod.rs

```diff
@@ -226,13 +223,14 @@ pub trait Jolt<F: PrimeField, G: CurveGroup<ScalarField = F>, const C: usize, co
         max_bytecode_size: usize,
         max_memory_address: usize,
         max_trace_length: usize,
-    ) -> JoltPreprocessing<F, G> {
-        let num_bytecode_generators =
-            BytecodePolynomials::<F, G>::num_generators(max_bytecode_size, max_trace_length);
-        let num_read_write_memory_generators =
-            ReadWriteMemory::<F, G>::num_generators(max_memory_address, max_trace_length);
-        let timestamp_range_check_generators =
-            TimestampValidityProof::<F, G>::num_generators(max_trace_length);
+    ) -> JoltPreprocessing<F, CS> {
+        let bytecode_generator_shapes =
+            BytecodePolynomials::<F, CS>::generator_shapes(max_bytecode_size, max_trace_length);
+        let ram_generator_shapes =
+            ReadWriteMemory::<F, CS>::generator_shapes(max_memory_address, max_trace_length);
+        let timestamp_range_check_generator_shapes =
+            TimestampValidityProof::<F, CS>::generator_shapes(max_trace_length);
```

Comment:
Maybe instead of these `generator_shapes` functions (which will end up being a misnomer for Binius), we should instead have `pcs_preprocess` functions which mutate a common `CS::Preprocessing` variable

Reply:
Switching to `GeneratorShape` -> `CommitShape`. Idea is that the commitments have a constant shape and then `PCS::setup(shapes: &[CommitShape]) -> PCS::Setup` can figure out how to do setup based on that. For EC commits this will be doing generator stuff, for hash based, no_op as you mention above.

In practice, we don't use the `StructuredCommitment` anymore so the `commitment_shape() -> Vec<CommitShape>` function on {bytecode, RAM, instruction lookups} should really just be on `VM`. I'm going to punt on that for now.